File: Recursive/maxDepthBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var maxDepth = function(root) {
    if(root == null){
      return 0;
    }
    let leftDepth = maxDepth(root.left);
    let rightDepth = maxDepth(root.right);
    let compare = leftDepth > rightDepth ? leftDepth + 1 : rightDepth + 1;
    return compare;
};

var maxDepth = function(root) {
  if(!root){
    return 0;
  }
  return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
};

File: Python Algos/maxDepthBinaryTree.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
        
class Solution:
    # Recursive DFS, easiest method
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
        
    # DFS
    def maxDepthIterative(self, root: Optional[TreeNode]) -> int:
        pass
    
    # Breath FS
    def maxDepthRecursive(self, root: Optional[TreeNode]) -> int:
        pass